lldb/source/Core/Module.cpp

```diff
@@ -1202,6 +1202,63 @@ void Module::ReportWarningToolchainMismatch(
     }
   }
 }
+
+bool Module::IsSwiftCxxInteropEnabled() {
+  switch (m_is_swift_cxx_interop_enabled) {
+  case eLazyBoolYes:
+    return true;
+  case eLazyBoolNo:
+    return false;
+  case eLazyBoolCalculate:
+    break;
+  }
+  EnableCxxInteropOpts interop_enabled =
+      Target::GetGlobalProperties().GetSwiftEnableCxxInteropOpts();
+  switch (interop_enabled) {
+  case eEnableCxxInterop:
+    m_is_swift_cxx_interop_enabled = eLazyBoolYes;
+    break;
+  case eDisableCxxInterop:
+    m_is_swift_cxx_interop_enabled = eLazyBoolNo;
+    break;
+  case eAutoDetectCxxInterop: {
+    // Heuristic: ignore anything that starts with "libswift", since the
+    // libraries shipped with Swift are written in C++.
+    auto name = GetFileSpec().GetFileNameStrippingExtension().GetStringRef();
+    if (name.startswith("libswift"))
+      return false;
```

Comment:
This a per-module function, so this would misfire for libSwiftCore.dSYM, which generally only people manually building their own Swift compiler would even have. Since developers might legitimately name libraries `libswift*` maybe it's better to accept a false positive inside the standard library for those few people that have debug info for it?